lib/rack/headers_filter.rb

```diff
@@ -0,0 +1,38 @@
+module Rack
+  class HeadersFilter
+    SENSITIVE_HEADERS = %w[
+      HTTP_X_FORWARDED_FOR
+      HTTP_X_FORWARDED_HOST
+      HTTP_X_FORWARDED_PORT
+      HTTP_X_FORWARDED_PROTO
+      HTTP_X_FORWARDED_SCHEME
+      HTTP_X_FORWARDED_SSL
+    ].freeze
+
+    # Headers sent by the Heroku router
+    HEROKU_HEADERS = %w[
```

Comment:
Are there any other headers we want/need to trust, e.g. coming from Fastly?

We may eventually also need to revisit this in the context of generalization where Heroku may not always be used.